spring-cloud-kubernetes-controllers/spring-cloud-kubernetes-discoveryserver/src/main/java/org/springframework/cloud/kubernetes/discoveryserver/DiscoveryCatalogWatcherController.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.discoveryserver;
+
+import java.util.List;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.kubernetes.commons.discovery.ConditionalOnKubernetesCatalogEnabled;
+import org.springframework.cloud.kubernetes.commons.discovery.EndpointNameAndNamespace;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author wind57
+ */
+@RestController
+@ConditionalOnKubernetesCatalogEnabled
+class DiscoveryCatalogWatcherController {
```

Comment:
this is a new and important thing. Actually my entire idea starts from here, from discovery server. 

- we already publish a heartbeat event from k8s reactive client
- discovery server already has the proper dependency, as it uses k8s reactive discovery client under the hood

As such, create a Listener for that event, make the listener a dependency in this controller, expose an endpoint that the http discovery client can periodically call and find out about changes in services, so that it can in turn publish proper heartbeat event. 

Also notice the `@ConditionalOnKubernetesCatalogEnabled` annotation, so that we are on par with the other two clients